src/NUnitFramework/framework/Constraints/CollectionContainsConstraint.cs

```diff
@@ -1,96 +0,0 @@
-// included in all copies or substantial portions of the Software.
-// 
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
-// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
-// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-// ***********************************************************************
-
-using System;
-using System.Collections;
-
-namespace NUnit.Framework.Constraints
-{
-    /// <summary>
-    /// CollectionContainsConstraint is used to test whether a collection
-    /// contains an expected object as a member.
-    /// </summary>
-    public class CollectionContainsConstraint : CollectionItemsEqualConstraint
-    {
-        /// <summary>
-        /// Construct a CollectionContainsConstraint
-        /// </summary>
-        /// <param name="expected"></param>
-        public CollectionContainsConstraint(object expected)
-            : base(expected)
-        {
-            Expected = expected;
-        }
-
-        /// <summary> 
-        /// The display name of this Constraint for use by ToString().
-        /// The default value is the name of the constraint with
-        /// trailing "Constraint" removed. Derived classes may set
-        /// this to another name in their constructors.
-        /// </summary>
-        public override string DisplayName { get { return "Contains"; } }
-
-        /// <summary>
-        /// The Description of what this constraint tests, for
-        /// use in messages and in the ConstraintResult.
-        /// </summary>
-        public override string Description
-        {
-            get { return "collection containing " + MsgUtils.FormatValue(Expected); }
-        }
-
-        /// <summary>
-        /// Gets the expected object
-        /// </summary>
-        protected object Expected { get; private set; }
-
-        /// <summary>
-        /// Test whether the expected item is contained in the collection
-        /// </summary>
-        /// <param name="actual"></param>
-        /// <returns></returns>
-        protected override bool Matches(IEnumerable actual)
-        {
-            foreach (object obj in actual)
-                if (ItemsEqual(obj, Expected))
-                    return true;
-
-            return false;
-        }
-
-
-        /// <summary>
-        /// Flag the constraint to use the supplied predicate function
-        /// </summary>
-        /// <param name="comparison">The comparison function to use.</param>
-        /// <returns>Self.</returns>
-        public CollectionContainsConstraint Using<TCollectionType, TMemberType>(Func<TCollectionType, TMemberType, bool> comparison)
```

Comment:
@ChrisMaddock, I think you are correct. @Mikkelbu, would you mind making an update to address this? Other than that, looks good.